src/main/java/com/morlunk/jumble/audio/AudioOutput.java

```diff
@@ -189,15 +189,28 @@ private boolean mix(short[] outBuffer) {
             }
 
             if(!mMixBuffer.isEmpty()) {
+
+                /* Declare temp vars here to avoid creation/destructions within loops */
+                short pcm = 0;
+                int pcm_tmp = 0;
+
                 for(AudioOutputSpeech.Result result : mMixBuffer) {
                     float[] buffer = result.getSamples();
                     for(int i = 0; i < BUFFER_SIZE; i++) {
-                        short pcm = (short) (buffer[i]*Short.MAX_VALUE); // Convert float to short
-                        pcm = pcm <= Short.MAX_VALUE ? (pcm >= Short.MIN_VALUE ? pcm : Short.MIN_VALUE) : Short.MIN_VALUE; // Clip audio
-                        outBuffer[i] += pcm;
+
+                        /* @TODO replace with higher quality mixing algorithm (better quality and maintaining loudness) */
+
+                        /* Convert float to short. float pcm values are in range [-1.0,1.0] */
+                        pcm = (short) (buffer[i]*Short.MAX_VALUE);
```

Comment:
I believe it is possible here for pcm to overflow; what if buffer[i] > 1f || buffer[i] < 1f?

It may be a better idea to mix the users' samples into a float buffer and then bound it appropriately after.